unit_tests/testHDF5FileType/testHDF5FileType.f90

```diff
@@ -1107,6 +1108,560 @@ SUBROUTINE testHDF5FileTypeWrite()
+      CALL h5%fwrite('groupST->memST3',refST3,SHAPE(refST3))
+      CALL h5%mkdir('groupC')
+      CALL h5%fwrite('groupC->memC1',refC1,LEN(refC1))
+
+      ! Write data second time, overwriting existing
+      CALL h5%mkdir('groupR')
+      CALL h5%fwrite('groupR->memD0',refD0)
+      CALL h5%fwrite('groupR->memD1',refD1,SHAPE(refD1))
+      CALL h5%fwrite('groupR->memD2',refD2,SHAPE(refD2))
+      CALL h5%fwrite('groupR->memD3',refD3,SHAPE(refD3))
+      CALL h5%fwrite('groupR->memD4',refD4,SHAPE(refD4))
+      CALL h5%fwrite('groupR->memD5',refD5,SHAPE(refD5))
+      CALL h5%fwrite('groupR->memD6',refD6,SHAPE(refD6))
+      CALL h5%fwrite('groupR->memD7',refD7,SHAPE(refD7))
+      CALL h5%fwrite('groupR->memS0',refS0)
+      CALL h5%fwrite('groupR->memS1',refS1,SHAPE(refS1))
+      CALL h5%fwrite('groupR->memS2',refS2,SHAPE(refS2))
+      CALL h5%fwrite('groupR->memS3',refS3,SHAPE(refS3))
+      CALL h5%fwrite('groupR->memS4',refS4,SHAPE(refS4))
+      CALL h5%fwrite('groupR->memS5',refS5,SHAPE(refS5))
+      CALL h5%fwrite('groupR->memS6',refS6,SHAPE(refS6))
+      CALL h5%fwrite('groupR->memS7',refS7,SHAPE(refS7))
+      CALL h5%mkdir('groupI')
+      CALL h5%fwrite('groupI->memL0',refL0)
+      CALL h5%fwrite('groupI->memL1',refL1,SHAPE(refL1))
+      CALL h5%fwrite('groupI->memL2',refL2,SHAPE(refL2))
+      CALL h5%fwrite('groupI->memL3',refL3,SHAPE(refL3))
+      CALL h5%fwrite('groupI->memL4',refL4,SHAPE(refL4))
+      CALL h5%fwrite('groupI->memL5',refL5,SHAPE(refL5))
+      CALL h5%fwrite('groupI->memL6',refL6,SHAPE(refL6))
+      CALL h5%fwrite('groupI->memL7',refL7,SHAPE(refL7))
+      CALL h5%fwrite('groupI->memN0',refN0)
+      CALL h5%fwrite('groupI->memN1',refN1,SHAPE(refN1))
+      CALL h5%fwrite('groupI->memN2',refN2,SHAPE(refN2))
+      CALL h5%fwrite('groupI->memN3',refN3,SHAPE(refN3))
+      CALL h5%fwrite('groupI->memN4',refN4,SHAPE(refN4))
+      CALL h5%fwrite('groupI->memN5',refN5,SHAPE(refN5))
+      CALL h5%fwrite('groupI->memN6',refN6,SHAPE(refN6))
+      CALL h5%fwrite('groupI->memN7',refN7,SHAPE(refN7))
+      CALL h5%mkdir('groupB')
+      CALL h5%fwrite('groupB->memB0',refB0)
+      CALL h5%fwrite('groupB->memB1',refB1,SHAPE(refB1))
+      CALL h5%fwrite('groupB->memB2',refB2,SHAPE(refB2))
+      CALL h5%fwrite('groupB->memB3',refB3,SHAPE(refB3))
+      CALL h5%mkdir('groupCNCHAR')
+      CALL h5%fwrite('groupCNCHAR->CNCHAR0',refCNCHAR0)
+      CALL h5%fwrite('groupCNCHAR->CNCHAR1',refCNCHAR1,SHAPE(refCNCHAR1))
+      CALL h5%fwrite('groupCNCHAR->CNCHAR2',refCNCHAR2,SHAPE(refCNCHAR2))
+      CALL h5%fwrite('groupCNCHAR->CNCHAR3',refCNCHAR3,SHAPE(refCNCHAR3))
+      CALL h5%mkdir('groupST')
+      CALL h5%fwrite('groupST->memST0',refST0)
+      CALL h5%fwrite('groupST->memST1',refST1,SHAPE(refST1))
+      CALL h5%fwrite('groupST->memST2',refST2,SHAPE(refST2))
+      CALL h5%fwrite('groupST->memST3',refST3,SHAPE(refST3))
+      CALL h5%mkdir('groupC')
+      CALL h5%fwrite('groupC->memC1',refC1,LEN(refC1))
+
```

Comment:
It doesn't appear that there are any asserts to ensure it actually changes the data.  I'd suggest writing some basic data (say 0's, then overwrite with the correct data and assert it actually changes them.